types.d.ts

```diff
@@ -0,0 +1,8715 @@
+	[index: string]: EntryDescriptionNormalized;
+}
+
+/**
+ * Entrypoint serves as an encapsulation primitive for chunks that are
+ * a part of a single ChunkGroup. They represent all bundles that need to be loaded for a
+ * single instance of a page. Multi-page application architectures will typically yield multiple Entrypoint objects
+ * inside of the compilation, whereas a Single Page App may only contain one with many lazy-loaded chunks.
+ */
+interface Entrypoint extends ChunkGroup {
+	runtimeChunk: Chunk;
+
+	/**
+	 * Sets the runtimeChunk for an entrypoint.
+	 */
+	setRuntimeChunk: (chunk: Chunk) => void;
+
+	/**
+	 * Fetches the chunk reference containing the webpack bootstrap code
+	 */
+	getRuntimeChunk: () => Chunk;
+}
+interface EnumParams {
+	allowedValues: Array<any>;
+}
+declare class EnvironmentPlugin {
+	constructor(...keys: Array<any>);
+	keys: Array<any>;
+	defaultValues: any;
+	apply: (compiler: Compiler) => void;
+}
+interface ErrorObject {
+	keyword: string;
+	dataPath: string;
+	schemaPath: string;
+	params:
+		| RefParams
+		| LimitParams
+		| AdditionalPropertiesParams
+		| DependenciesParams
+		| FormatParams
+		| ComparisonParams
+		| MultipleOfParams
+		| PatternParams
+		| RequiredParams
+		| TypeParams
+		| UniqueItemsParams
+		| CustomParams
+		| PatternRequiredParams
+		| PropertyNamesParams
+		| IfParams
+		| SwitchParams
+		| NoParams
+		| EnumParams;
+	propertyName?: string;
+	message?: string;
+	schema?: any;
+	parentSchema?: {};
+	data?: any;
+}
+interface Etag {
+	toString: () => string;
+}
+declare class EvalDevToolModulePlugin {
+	constructor(options: any);
+	namespace: any;
+	sourceUrlComment: any;
+	moduleFilenameTemplate: any;
+	apply: (compiler: Compiler) => void;
+}
+declare class EvalSourceMapDevToolPlugin {
+	constructor(options: any);
+	sourceMapComment: any;
+	moduleFilenameTemplate: any;
+	namespace: any;
+	options: any;
+	apply: (compiler: Compiler) => void;
+}
+
+/**
+ * Enables/Disables experiments (experimental features with relax SemVer compatibility).
+ */
+interface Experiments {
+	/**
+	 * Allow module type 'asset' to generate assets.
+	 */
+	asset?: boolean;
+
+	/**
+	 * Support WebAssembly as asynchronous EcmaScript Module.
+	 */
+	asyncWebAssembly?: boolean;
+
+	/**
+	 * Allow 'import/export' syntax to import async modules.
+	 */
+	importAsync?: boolean;
+
+	/**
+	 * Allow 'import/export await' syntax to import async modules.
+	 */
+	importAwait?: boolean;
+
+	/**
+	 * Support .mjs files as way to define strict ESM file (node.js).
+	 */
+	mjs?: boolean;
+
+	/**
+	 * Allow output javascript files as module source type.
+	 */
+	outputModule?: boolean;
+
+	/**
+	 * Support WebAssembly as synchronous EcmaScript Module (outdated).
+	 */
+	syncWebAssembly?: boolean;
+
+	/**
+	 * Allow using top-level-await in EcmaScript Modules.
+	 */
+	topLevelAwait?: boolean;
+}
+interface ExportInfo {
+	name: string;
+	usedName: string | typeof SKIP_OVER_NAME;
+	used: 0 | 1 | 2 | 3 | 4;
```

Comment:
@sokra Do I get it right - all simple typedefs inlined ? Enum here inlined too..